src/libNetwork/DataSender.cpp

```diff
@@ -21,40 +21,34 @@
 #include "libNetwork/Blacklist.h"
 #include "libNetwork/P2PComm.h"
 #include "libUtils/DataConversion.h"
+#include "libUtils/DnsUtils.h"
 #include "libUtils/IPConverter.h"
 #include "libUtils/Logger.h"
 
 using namespace std;
 
-void SendDataToLookupNodesDefault(const VectorOfNode& lookups,
-                                  const bytes& message) {
+void SendDataToLookupNodesDefault(const bytes& message) {
   if (LOOKUP_NODE_MODE) {
     LOG_GENERAL(WARNING,
                 "DataSender::SendDataToLookupNodesDefault not "
                 "expected to be called from LookUp node.");
   }
   LOG_MARKER();
 
-  vector<Peer> allLookupNodes;
+  vector<string> ipList;
 
-  for (const auto& node : lookups) {
-    string url = node.second.GetHostname();
-    auto resolved_ip = node.second.GetIpAddress();  // existing one
-    if (!url.empty()) {
-      uint128_t tmpIp;
-      if (IPConverter::ResolveDNS(url, node.second.GetListenPortHost(),
-                                  tmpIp)) {
-        resolved_ip = tmpIp;  // resolved one
-      } else {
-        LOG_GENERAL(WARNING, "Unable to resolve DNS for " << url);
-      }
-    }
+  ObtainIpListFromDns(ipList, LOOKUPS_URL);
```

Comment:
I think we want to avoid obtaining IP list every time we have to send a message. Better to still retain local list and only obtain from DNS server at pre-determined intervals (e.g., new DS epoch)